### PR TITLE
Finish Service Architecture Setup

### LINKED ISSUE #3: Service architecture design

### PR DESCRIPTION
**Pull request by Alex-XJK**
This is the first architecture set up, led by @Alex-XJK 
- Completed hand-drawn service architecture design
- Completed the logical walkthrough with @phoebeww 
- Implemented the necessary Java Classes and Interfaces
- Applied Command Design Patterns, Singleton Design Patterns, Facade Design Patterns, and Composite Design Patterns
- Provided adequate Javadoc comments for colleagues to continue development
- Proposed ideas for follow-up development and demo.

### CODE REVIEW

**Comment by Alex-XJK** [comment]
Class UML diagram as of commit 4454f4e.
project

**Comment by Alex-XJK** [comment]
Given that this is the most basic and critical service architecture, all members need to review the code and provide feedback. After everyone approves without objection, I will proceed with the merge operation.

When this PR gets merged, it will close #3

**Review by phoebeww — APPROVED** [review]
Looks good to me, Thank you Alex

File: 100of100Service/src/main/java/dev/coms4156/project/Employee.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+   * @param db the HR database facade that manages the employee
+   * @param id the ID of the employee
+   * @param name the name of the employee
+   * @param hireDate the hire date of the employee
+   */
+  public Employee(HRDatabaseFacade db, int id, String name, Date hireDate) {
+    this.db = db;
+    this.id = id;
+    this.name = name;
+    this.hireDate = hireDate;
```
Comment (sophiefeifeifeiya) [inline 100of100Service/src/main/java/dev/coms4156/project/Employee.java]: this.hireDate = new Date(hireDate.getTime());

File: 100of100Service/src/main/java/dev/coms4156/project/Employee.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  public List<OrganizationComponent> getChildren() {
+    return new ArrayList<>();
+  }
+
+  /**
+   * Returns the hire date of the employee.
+   * @return the hire date of the employee
+   */
+  public Date getHireDate() {
+    return this.hireDate;
```
Comment (sophiefeifeifeiya) [inline 100of100Service/src/main/java/dev/coms4156/project/Employee.java]: return new Date(this.hireDate.getTime());

**Pull request merged**